buildSrc/src/main/groovy/org/apache/fineract/gradle/service/GpgService.groovy

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.gradle.service
+
+import org.apache.fineract.gradle.FineractPluginExtension
+import org.apache.fineract.gradle.FineractPluginExtension.FineractPluginConfigGpg
+import org.bouncycastle.bcpg.ArmoredOutputStream
+import org.bouncycastle.bcpg.BCPGOutputStream
+import org.bouncycastle.gpg.keybox.BlobType
+import org.bouncycastle.gpg.keybox.PublicKeyRingBlob
+import org.bouncycastle.gpg.keybox.bc.BcKeyBox
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.openpgp.*
+import org.bouncycastle.openpgp.operator.bc.BcKeyFingerprintCalculator
+import org.bouncycastle.openpgp.operator.bc.BcPGPContentSignerBuilder
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.security.GeneralSecurityException
+import java.security.MessageDigest
+import java.security.Security
+
+class GpgService {
+    private static final Logger log = LoggerFactory.getLogger(GpgService.class)
+
+    private PGPPrivateKey privateKey;
+    private PGPPublicKey publicKey;
+
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    GpgService(FineractPluginConfigGpg config) {
+        // TODO: provide implementation for *.gpg key rings
```

Comment:
Ticket maybe if we need to do it later?

Reply:
*.gpg format doesn't seem to be the standard anymore. So this is more like an optional personal reminder.